simulation/agent/ParenthoodAgent.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.vaticle.typedb.benchmark.simulation.agent;
+
+import com.vaticle.typedb.benchmark.common.concept.Country;
+import com.vaticle.typedb.benchmark.common.concept.Marriage;
+import com.vaticle.typedb.benchmark.common.concept.Parenthood;
+import com.vaticle.typedb.benchmark.common.concept.Person;
+import com.vaticle.typedb.benchmark.common.params.Context;
+import com.vaticle.typedb.benchmark.common.seed.RandomSource;
+import com.vaticle.typedb.benchmark.simulation.driver.Client;
+import com.vaticle.typedb.benchmark.simulation.driver.Session;
+import com.vaticle.typedb.benchmark.simulation.driver.Transaction;
+import com.vaticle.typedb.common.collection.Pair;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.vaticle.typedb.common.collection.Collections.list;
+import static java.util.Comparator.comparing;
+import static java.util.stream.Collectors.toCollection;
+
+public abstract class ParenthoodAgent<TX extends Transaction> extends Agent<Country, TX> {
```

Comment:
Surely that goes against the agent-based model? In terms of benchmarking also this would be a lot more work done in a single agent for readers to unpack. Let's discuss